packages/components/date-picker/src/date-picker-com/panel-year-range.vue

```diff
@@ -0,0 +1,264 @@
+import { useYearRangeHeader } from '../composables/use-year-range-header'
+import { isValidRange } from '../utils'
+import { ROOT_PICKER_INJECTION_KEY } from '../constants'
+import YearTable from './basic-year-table.vue'
+
+import type { Dayjs } from 'dayjs'
+import type { RangeState } from '../props/shared'
+
+defineOptions({
+  name: 'DatePickerYearRange',
+})
+
+const props = defineProps(panelYearRangeProps)
+const emit = defineEmits(panelYearRangeEmits)
+
+const { lang } = useLocale()
+const leftDate = ref(dayjs().locale(lang.value))
+const rightDate = ref(leftDate.value.add(10, 'year'))
+const { pickerNs: ppNs } = inject(ROOT_PICKER_INJECTION_KEY)!
+const drpNs = useNamespace('date-range-picker')
+
+const hasShortcuts = computed(() => !!shortcuts.length)
+
+const panelKls = computed(() => [
+  ppNs.b(),
+  drpNs.b(),
+  {
+    'has-sidebar': Boolean(useSlots().sidebar) || hasShortcuts,
+  },
+])
+
+const handleShortcutClick = useShortcut(lang)
+
+const {
+  leftPrevYear,
+  rightNextYear,
+  leftNextYear,
+  rightPrevYear,
+  leftLabel,
+  rightLabel,
+  leftYear,
+  rightYear,
+} = useYearRangeHeader({
+  unlinkPanels: toRef(props, 'unlinkPanels'),
+  leftDate,
+  rightDate,
+})
+
+const enableYearArrow = computed(() => {
+  return props.unlinkPanels && rightYear.value > leftYear.value + 1
+})
+
+const minDate = ref<Dayjs>()
+const maxDate = ref<Dayjs>()
+
+const rangeState = ref<RangeState>({
+  endDate: null,
+  selecting: false,
+})
+
+const handleChangeRange = (val: RangeState) => {
+  rangeState.value = val
+}
+
+type RangePickValue = {
+  minDate: Dayjs
+  maxDate: Dayjs
+}
+const handleRangePick = (val: RangePickValue, close = true) => {
+  const minDate_ = val.minDate
+  const maxDate_ = val.maxDate
+  if (maxDate.value === maxDate_ && minDate.value === minDate_) {
+    return
+  }
+  emit('calendar-change', [minDate_.toDate(), maxDate_ && maxDate_.toDate()])
+  maxDate.value = maxDate_
+  minDate.value = minDate_
+
+  if (!close) return
+  handleConfirm()
+}
+
+const handleConfirm = (visible = false) => {
+  if (isValidRange([minDate.value, maxDate.value])) {
+    emit('pick', [minDate.value, maxDate.value], visible)
+  }
+}
+
+const onSelect = (selecting: boolean) => {
+  rangeState.value.selecting = selecting
+  if (!selecting) {
+    rangeState.value.endDate = null
+  }
+}
+
+const formatToString = (days: Dayjs[]) => {
+  return days.map((day) => day.format(format))
+}
+const getDefaultValue = () => {
+  let start: Dayjs
+  if (isArray(defaultValue.value)) {
+    const left = dayjs(defaultValue.value[0])
+    let right = dayjs(defaultValue.value[1])
+    if (!props.unlinkPanels) {
+      right = left.add(10, 'year')
+    }
+    return [left, right]
+  } else if (defaultValue.value) {
+    start = dayjs(defaultValue.value)
+  } else {
+    start = dayjs()
+  }
+  start = start.locale(lang.value)
+  return [start, start.add(10, 'year')]
+}
+
+emit('set-picker-option', ['formatToString', formatToString])
```

Comment:
We need to refer to the `panel-date-range.vue` to improve more methods.

```ts
emit('set-picker-option', ['isValidValue', isValidValue])
emit('set-picker-option', ['parseUserInput', parseUserInput])
emit('set-picker-option', ['formatToString', formatToString])
emit('set-picker-option', ['handleClear', handleClear])
```

Currently, an error will be reported after manually entering the time in the input box